spring-cloud-gcp-pubsub-stream-binder/src/test/java/org/springframework/cloud/gcp/stream/binder/pubsub/PubSubMessageChannelBinderTests.java

```diff
@@ -44,14 +44,15 @@
 
 	@BeforeClass
 	public static void enableTests() {
-		assumeTrue("PubSub Binder tests are disabled. Please enable them with -Dit.pubsub or -Dit.localdeps",
-				"true".equals(System.getProperty("it.pubsub")) || "true".equals(System.getProperty("it.localdeps")));
-		assumeThat(emulator.getEmulatorHostPort())
-				.withFailMessage("No PubSub emulator host configured").isNotNull();
+		assumeTrue("PubSub Binder tests are disabled. Please enable them with -Dit.pubsub or -Dit.emulator",
```

Comment:
I still don't see reason in overhead with this method. 
i just don't want to copy/paste always together with the `PubSubEmulator` rule.
All the logic simply can be done in the rule generic way.
And it already doesn't matter what consumer test-case is for this rule.

Reply:
Do you mean that the gating flag could be consumed by PubSubEmulator rule itself?

Reply:
Correct. Otherwise the rule just does not make sense. I feel them as together: if we don't set flag, we don't go to check any other options.
More over as you noticed the `@ClassRule` works before this local `@ClassRule`, but that's not we would like to have when we don't set the system flag.

Hope this facilitates a bit my other concern about `getBinder()`  body.